mindmeld/rasaconverter.py

```diff
@@ -0,0 +1,545 @@
+
+# The namespace of the application. Used to prevent collisions in supporting services across
+# applications. If not set here, the app's enclosing directory name is used.
+# APP_NAMESPACE = 'app-name'
+
+# Dictionaries for the various NLP classifier configurations
+
+# An example decision tree model for intent classification
+INTENT_CLASSIFIER_CONFIG = {
+    'model_type': 'text',
+    'model_settings': {
+        'classifier_type': 'dtree'
+    },
+    'param_selection': {
+        'type': 'k-fold',
+        'k': 10,
+        'grid': {
+            'max_features': ['log2', 'sqrt', 0.01, 0.1]
+        },
+    },
+    "features": {
+        "exact": {},
+    }
+}
+
+"""
+# Fill in the other model configurations if necessary
+# DOMAIN_CLASSIFIER_CONFIG = {}
+# ENTITY_RECOGNIZER_CONFIG = {}
+# ROLE_CLASSIFIER_CONFIG = {}
+"""
+
+# A example configuration for the parser
+"""
+# *** Note: these are place holder entity types ***
+PARSER_CONFIG = {
+    'grandparent': {
+        'parent': {},
+        'child': {'max_instances': 1}
+    },
+    'parent': {
+        'child': {'max_instances': 1}
+    }
+}
+"""'''
+        with open(mindmeld_project_directory + "/config.py", "w") as f:
+            f.write(string)
+
+    @staticmethod
+    def create_custom_features(mindmeld_project_directory):
+        string = '''from mindmeld.models.helpers import register_query_feature, register_entity_feature
+
+
+@register_query_feature(feature_name='average-token-length')
+def extract_average_token_length(**args):
+    """
+    Example query feature that gets the average length of normalized tokens in the query
+
+    Returns:
+        (function) A feature extraction function that takes a query and
+            returns the average normalized token length
+    """
+    def _extractor(query, resources):
+        tokens = query.normalized_tokens
+        average_token_length = sum([len(t) for t in tokens]) / len(tokens)
+        return {'average_token_length': average_token_length}
+
+    return _extractor
+
+
+@register_entity_feature(feature_name='entity-span-start')
+def extract_entity_span_start(**args):
+    """
+    Example entity feature that gets the start span for each entity
+
+    Returns:
+        (function) A feature extraction function that returns the start span of the entity
+    """
+    def _extractor(example, resources):
+        query, entities, entity_index = example
+        features = {}
+
+        current_entity = entities[entity_index]
+        current_entity_token_start = current_entity.token_span.start
+
+        features['entity_span_start'] = current_entity_token_start
+        return features
+
+    return _extractor'''
+        with open(mindmeld_project_directory + "/custom_features.py", "w") as f:
+            f.write(string)
+
+    def convert_project(self):
+        # Create project directory with sub folders
+        Rasaconverter.create_mindmeld_directory(self.mindmeld_project_directory)
```

Comment:
you can also do `self. create_mindmeld_directory(self.mindmeld_project_directory)`